newIDE/app/src/MainFrame/index.js

```diff
@@ -521,14 +534,29 @@ const MainFrame = (props: Props) => {
 
   React.useEffect(
```

Comment:
Can I suggest that this effect goes in its own custom hook to lighten a bit MainFrame?

Reply:
And its own state variables

Reply:
Good idea, some "useInitialState" or something like this!